qiskit/quantum_info/mitigation/complete_readout_mitigator.py

```diff
@@ -0,0 +1,275 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Readout mitigator class based on the A-matrix inversion method
+"""
+
+import logging
+from typing import Optional, List, Dict, Tuple, Iterable, Callable
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.result import Counts, marginal_counts, QuasiDistribution
+from .base_readout_mitigator import BaseReadoutMitigator
+
+logger = logging.getLogger(__name__)
+
+
+class CompleteReadoutMitigator(BaseReadoutMitigator):
+    """N-qubit readout error mitigator.
+    Mitigates expectation_value and quasi_probabilities.
+    The mitigation_matrix should be calibrated using qiskit.experiments."""
+
+    def __init__(self, amat: np.ndarray):
+        """Initialize a CompleteReadoutMitigator
+        Args:
+            amat (np.array): readout error assignment matrix.
+        """
+        self._num_qubits = int(np.log2(amat.shape[0]))
+        self._assignment_mat = amat
+        self._mitigation_mats = {}
+
+    def expectation_value(
+        self,
+        data: Counts,
+        diagonal: Callable,
+        qubits: Iterable[int] = None,
+        clbits: Optional[List[int]] = None,
```

Comment:
Clbits isn't in the base class. It should either be added there, or removed from here.

In this case the point of clbits kwarg in this method is to allow marginalizing the input counts so you don't have to define a diagonal on the full set of qubits. You could equivalently do something like that with a callable but its a bit more work on the user

Reply:
which of the options is better?